subsys/bluetooth/mesh/net.c

```diff
@@ -624,7 +625,7 @@ static void bt_mesh_net_relay(struct net_buf_simple *sbuf,
 	 * Anything else (like GATT to adv, or locally originated packets)
 	 * use the Network Transmit state.
 	 */
-	if (rx->net_if == BT_MESH_NET_IF_ADV) {
+	if (rx->net_if == BT_MESH_NET_IF_ADV && !rx->friend_match) {
```

Comment:
Could you explain this change a bit more? The protocol between LPN and Friend is timing and request-response based. Is there somewhere in the spec that says that it should use the RelayRetransmit state rather than the NetworkTransmit state?

Reply:
Please see Mesh Specification 1.0.1 Section 3.6.6.2 Friendship security for more information.
`The Low Power node then decides to send two messages: OutMsg1 and OutMsg2. OutMsg1 is sentsecured using the friend security credentials and therefore only the Friend node will receive and relay thismessage. When the Friend node relays OutMsg1, the message will be retransmitted using the mastersecurity credentials.`